### PR TITLE
Implement UI library as a submodule

### PR DESCRIPTION
**Pull request by jthrilly**
This PR implements the UI library as a submodule within Network Canvas. This implementation required:

- Updating paths and imports that reference UI components and styles
- Implementing the `$font-path` variable to provide a relative font path. We tried using a relative path directly in the UI submodule, but couldn't get this to work.
- Removing the webpack alias for the font path (we could have used this instead of the above, but we felt it created an odd dependency).
- Updating the package.json to remove the UI dep
- Updating the sass linting rules to allow the combined NC + UI codebase to lint

Running this branch will require an `npm install` and a `git submodule update --recursive`.

For now, this branch is tracking the `feature/submodule` branch of UI (https://github.com/codaco/Network-Canvas-UI/pull/53). Once we are happy with the implementation, I will merge that branch, and update the ref here.

### CODE REVIEW

**Comment by jthrilly** [comment]
@wwqrd - do you mind checking if the Travis configuration looks ok? I tried to add a sass-lint task to the lint stage following the build stages docs, but I'm worried that only the second task ran.

File: .sass-lint.yml [hunk 1]
```diff
@@ -46,15 +46,15 @@ rules:
 
   # Nesting
   declarations-before-nesting: 1
-  force-attribute-nesting: 1
-  force-element-nesting: 1
-  force-pseudo-nesting: 1
+  force-attribute-nesting: 0
```
[the remark this answers is not in the document]
Comment (jthrilly) [reply .sass-lint.yml]: > My two cents: these should be on. NC-UI is unexpectedly specific in places, which required workarounds in Server; having these warnings forces you to think about it.

I might be misunderstanding, but I think these rules, couple with BEM, are actually the cause the our high specificity. They are designed to not allow leaky rules. 

Regardless, I've turned these back on and fixed the errors, as you suggested. I've also added specific linting to UI which is currently the same as NC, but which can be adapted if we need to.

File: package-lock.json [hunk 2]
```diff
@@ -10,32 +10,16 @@
             "integrity": "sha512-juYJNi8JEpTUWXwz8ssa8Oop4n/kwJ/pIQP22vJAVAe6RTRD+0m+e9LRNnfK2EDaX8uwmUzLNGviFQRD6SxeOw==",
             "dev": true,
             "requires": {
-                "7zip-bin-linux": "1.3.1",
-                "7zip-bin-mac": "1.0.1",
-                "7zip-bin-win": "2.2.0"
+                "7zip-bin-mac": "1.0.1"
```
[the remark this answers is not in the document]
Comment (jthrilly) [reply package-lock.json]: Gah. I was having trouble with my version manager. I will recreate this.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,6 @@
 [submodule "src/utils/networkQuery"]
 	path = src/utils/networkQuery
 	url = https://github.com/codaco/networkQuery
+[submodule "src/ui"]
```
[the remark this answers is not in the document]
Comment (jthrilly) [reply .gitmodules]: Steve and I talked about this today and came to the conclusion that "src" made more sense here. 

- Using lib would mean a webpack alias, which would mean two points of configuration
- This isn't really a library, in the traditional sense anyway
- If we were going to go by "role" we would have put networkQuery somewhere else (src/externals, or something).
- We can document this, and change it later if it becomes an issue.

**Comment by jthrilly** [comment]
Okay, the UI submodule now has its own listing, but it doesn't yet have tests. I think we should add these in another PR since it is out of scope for here.

Other than that, I have made some small CSS tweaks that were outstanding. 

@bryfox - any further changes you want to see?
@rebeccamadsen - is this workflow working on windows?

**Comment by rebeccamadsen** [comment]
Turning off that linebreak-style as a temporary work-around, things look like they are working mostly! So far I've only noticed two things beyond the linebreak issue.

1. The side panels aren't collapsing. The nodes disappear when you click on the panel title, but the panels still take up the same height. 
2. And part of the toggle in the education protocol's node form kind of blends in with the background color of the modal version of the form when it is unselected.

**Comment by jthrilly** [comment]
> And part of the toggle in the education protocol's node form kind of blends in with the background color of the modal version of the form when it is unselected.

I've caught this, but it is part of a broader design issue where the full screen modal and the normal modal have different background colors. You'd think it would be as easy as picking the color that works for both, but neither seem to! Suggestions welcome. In the meantime, I had planned to fix this by making the background of the toggle a neutral gray.



> The side panels aren't collapsing. The nodes disappear when you click on the panel title, but the panels still take up the same height.

This will be a bug after refactoring. Thanks.

**Comment by jthrilly** [comment]
> I think we need a .gitattributes file to match what's in NC, maybe?

I've added this, too.

File: package-lock.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bryfox) [reply package-lock.json]: @jthrilly looks like .nvmrc has been removed... maybe that caused this? Or did you intentionally remove it to fix whatever nvm problems you were having?

**Pull request merged**